File: dsa_leetcode_beginner/0191-number-of-1-bits/0191-number-of-1-bits.cpp

#include <iostream>

using namespace std;

int hammingWeight(uint32_t n) {
    int count = 0;

    // Assuming a 32-bit integer
    for (int i = 0; i < 32; i++) {
        count += (n & 1);
        n = n >> 1;
    }

    return count;
}

int main() {
    uint32_t n = 11;  // Example input (binary: 1011)
    cout << hammingWeight(n) << endl;  // Output: 3

    return 0;
}

File: dsa_leetcode_beginner/0191-number-of-1-bits/0191-number-of-1-bits.js

function hammingWeight(n) {
    let count = 0;

    // Assuming a 32-bit integer
    for (let i = 0; i < 32; i++) {
        count += (n & 1);
        n = n >> 1;
    }

    return count;
}

// Example usage:
let n = 11;  // Example input (binary: 1011)
console.log(hammingWeight(n));  // Output: 3
